cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java

```diff
@@ -22,9 +23,13 @@
 
     PdfReportBuilder addElement(ReportElement element);
 
-    PdfReportBuilder withHeaderOnAllPages(String left, String middle, String right);
+    PdfReportBuilder withHeaderOnAllPages(String left, String middle, String right, boolean excludeOnAllPages);
```

Comment:
here you call the new boolean parameter `excludeOnAllPages` in the impl `excludeOnFirstPage` - which one is it now?

also, once again, this is an interface change. and again the usage is not intuitive, the method is called `OnAllPages` but the last parameter builds an exception to that so it wouldb e `OnAllPagesButTheFirst` ?
would you like to use such an API?

please always use names that actually tell you what is going to happen. also avoid boolean switches.
follow guidelines like the `Clean Code` book by Robert Cecil Martin

also it isn't extensible any more - imagine in your next update you want to allow to add headers only for odd pages. how would you accomplish that?

i would very much like something like this:

- leave the current signature like it is, it doesn't make sense to remove it, just provide an overloaded function and call it with the appropriate defaults (if necessary)
- provide either new functions (appropriately named) like `withHeaderOnAllPagesButFirst` (yes names are allowed to be long!) 
- or provide a more generic api like `withHeaderOnPagesStartingFrom` which accepts a `int startPage` (you could also provide stuff like `withHeaderOnPagesInRange` to have and endPage as well, etc.)
- or another generic approach like `withHeaderOnPages` which accepts an enum with certain values eg. `ALL`, `ALL_BUT_FIRST`, 'ALL_BUT_LAST`, 'EVEN', 'ODD', etc. - this list is exensible, if not dynamic
- or the most generic (and therefor rather cumbersome to use) approach:  `withHeaderOnPages` which accepts a List/Collection/Set/Iterable of page numbers